packages/coding-standards/src/Sniffs/ConstantVisibilityRequiredSniff.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\CodingStandards\Sniffs;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+use PHP_CodeSniffer\Util\Tokens;
+
+class ConstantVisibilityRequiredSniff implements Sniff
+{
+    /**
+     * @return int[]
+     */
+    public function register(): array
+    {
+        return [\T_CONST];
+    }
+
+    /**
+     * @param \PHP_CodeSniffer\Files\File $file
+     * @param $constPosition
+     */
+    public function process(File $file, $constPosition): void
+    {
+        if (!$this->isConstInsideClass($file, $constPosition)) {
+            return;
+        }
+
+        $previousTokenEndPosition = $file->findPrevious([\T_SEMICOLON, \T_CLOSE_CURLY_BRACKET], $constPosition) ?: 0;
+
+        if ($this->isConstWithAccessModifier($file, $constPosition, $previousTokenEndPosition)) {
+            return;
+        }
+
+        if ($this->isConstWithAccessAnnotation($file, $constPosition, $previousTokenEndPosition)) {
+            return;
+        }
+
+        $file->addError('Constant must have access modifier', $constPosition, self::class);
+    }
+
+    /**
+     * @param \PHP_CodeSniffer\Files\File $file
+     * @param int $constPosition
+     * @return bool
+     */
+    private function isConstInsideClass(File $file, int $constPosition): bool
+    {
+        $classStartPosition = $file->findPrevious(\T_CLASS, $constPosition);
+        if ($classStartPosition === false) {
+            return false;
+        }
+
+        $tokens = $file->getTokens();
+        $classEndPosition = $tokens[$classStartPosition]['scope_closer'];
+
+        return $constPosition > $classStartPosition && $constPosition < $classEndPosition;
+    }
+
+    /**
+     * @param \PHP_CodeSniffer\Files\File $file
+     * @param int $constPosition
+     * @param int $previousTokenEndPosition
+     * @return bool
+     */
+    private function isConstWithAccessModifier(File $file, int $constPosition, int $previousTokenEndPosition): bool
+    {
+        $accessModifierStartPosition = $file->findPrevious(Tokens::$scopeModifiers, $constPosition, $previousTokenEndPosition);
+
+        return (bool)$accessModifierStartPosition;
+    }
+
+    /**
+     * @param \PHP_CodeSniffer\Files\File $file
+     * @param int $constPosition
+     * @param int $previousTokenEndPosition
+     * @return bool
+     */
+    private function isConstWithAccessAnnotation(File $file, int $constPosition, int $previousTokenEndPosition): bool
+    {
+        $phpDocStartPosition = $file->findPrevious(\T_DOC_COMMENT_OPEN_TAG, $constPosition, $previousTokenEndPosition ?: 0);
+
+        return $phpDocStartPosition !== false && $this->phpDocContainsAccessTag($file, $phpDocStartPosition);
```

Comment:
It's difficult to read for me

```suggestion
        if ($phpDocStartPosition === false) {
            return false;
        }
        return $this->phpDocContainsAccessTag($file, $phpDocStartPosition);
```